[2024-06-20T07:16:28.771+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-06-20T07:16:28.794+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: ml_pipeline.preprocess_train_data manual__2024-06-20T07:16:24.217978+00:00 [queued]>
[2024-06-20T07:16:28.802+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: ml_pipeline.preprocess_train_data manual__2024-06-20T07:16:24.217978+00:00 [queued]>
[2024-06-20T07:16:28.803+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 2
[2024-06-20T07:16:28.821+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): preprocess_train_data> on 2024-06-20 07:16:24.217978+00:00
[2024-06-20T07:16:28.826+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61: DeprecationWarning: This process (pid=72) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-06-20T07:16:28.828+0000] {standard_task_runner.py:63} INFO - Started process 76 to run task
[2024-06-20T07:16:28.828+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'ml_pipeline', 'preprocess_train_data', 'manual__2024-06-20T07:16:24.217978+00:00', '--job-id', '27', '--raw', '--subdir', 'DAGS_FOLDER/ml_pipeline.py', '--cfg-path', '/tmp/tmp0x48rpzy']
[2024-06-20T07:16:28.829+0000] {standard_task_runner.py:91} INFO - Job 27: Subtask preprocess_train_data
[2024-06-20T07:16:28.882+0000] {task_command.py:426} INFO - Running <TaskInstance: ml_pipeline.preprocess_train_data manual__2024-06-20T07:16:24.217978+00:00 [running]> on host 9e16cb9136e3
[2024-06-20T07:16:28.972+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='RCH4CKERS' AIRFLOW_CTX_DAG_ID='ml_pipeline' AIRFLOW_CTX_TASK_ID='preprocess_train_data' AIRFLOW_CTX_EXECUTION_DATE='2024-06-20T07:16:24.217978+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-06-20T07:16:24.217978+00:00'
[2024-06-20T07:16:28.974+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-06-20T07:16:28.991+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-06-20T07:16:28.992+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/ml_pipeline.py", line 153, in preprocess_train_data
    data = process_data(data)
           ^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/ml_pipeline.py", line 34, in process_data
    numerical = process_numerical_df(numerical)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/ml_pipeline.py", line 52, in process_numerical_df
    import sklearn.preprocessing as preprocessing
ModuleNotFoundError: No module named 'sklearn'
[2024-06-20T07:16:29.005+0000] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=ml_pipeline, task_id=preprocess_train_data, run_id=manual__2024-06-20T07:16:24.217978+00:00, execution_date=20240620T071624, start_date=20240620T071628, end_date=20240620T071629
[2024-06-20T07:16:29.019+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 27 for task preprocess_train_data (No module named 'sklearn'; 76)
[2024-06-20T07:16:29.044+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-06-20T07:16:29.054+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
